src/D2L.Bmx/UpdateChecker.cs

```diff
@@ -89,19 +91,18 @@ internal class UpdateChecker( IGitHubClient github, BmxConfig config ) {
 		}
 	}
 
-	private static bool ShouldFetchLatestVersion( UpdateCheckCache? cache ) {
-		if( cache is null || string.IsNullOrWhiteSpace( cache.VersionName )
-			|| ( DateTimeOffset.UtcNow - cache.TimeLastChecked ) > TimeSpan.FromDays( 1 )
-			|| ( cache.TimeLastChecked > DateTimeOffset.UtcNow )
-		) {
-			return true;
-		}
-		return false;
+	private static bool ShouldFetchLatestVersion(
+		[NotNullWhen( returnValue: false )] UpdateCheckCache? cache
+	) {
+		return cache?.VersionName is null
+			|| cache.TimeLastChecked is null
+			|| DateTimeOffset.UtcNow - cache.TimeLastChecked > TimeSpan.FromDays( 1 )
+			|| cache.TimeLastChecked > DateTimeOffset.UtcNow;
 	}
 }
 
 
-internal record UpdateCheckCache {
-	public string? VersionName { get; set; }
```

Comment:
* These properties should not be settable
* `Version` can be deserialized to from JSON directly.

Reply:
How did you figure that out re: Version, at least as far as I can tell VSCode, Resharper etc.. don't inform of the fact it has a Serializable annotation.

Reply:
types don't need serializable annotation to be JSON serializable.
also, easy enough just to try with https://github.com/waf/CSharpRepl
![image](https://github.com/Brightspace/bmx/assets/8552945/4d7a34ce-ad2f-421b-8e5b-ea1e56bed565)